### ISSUE TITLE
请问VQVAE(stage1)阶段是怎样使用多级VectorQuantizer的?

### ISSUE THREAD

**Issue by YilanWang** [FoundationVision/VAR#42]
感谢作者开源的代码,我发现paper的algorithm 2里写了是把z_k随着分辨率升高,也一直插值,然后一起送入decoder里,但是我看代码,decoder好像是直接那z_k使用,并没有相关的插值啊? 
`

        class Decoder(nn.Module):
        
            def forward(self, z):
                # z to block_in
                # middle
                h = self.mid.block_2(self.mid.attn_1(self.mid.block_1(self.conv_in(z))))
                
        
                for i_level in reversed(range(self.num_resolutions)):
                    for i_block in range(self.num_res_blocks + 1):
                        h = self.up[i_level].block[i_block](h)
                        if len(self.up[i_level].attn) > 0:
                            h = self.up[i_level].attn[i_block](h)
                    if i_level != 0:
                        h = self.up[i_level].upsample(h)
                
        
                h = self.conv_out(F.silu(self.norm_out(h), inplace=True))
                return h

`

然后相应的upsample函数也是直接的上采样:
`

        class Upsample2x(nn.Module):
            def __init__(self, in_channels):
                super().__init__()
                self.conv = torch.nn.Conv2d(in_channels, in_channels, kernel_size=3, stride=1, padding=1)
            
            def forward(self, x):
        return self.conv(F.interpolate(x, scale_factor=2, mode='nearest'))

`

请问下algorithm2是怎么体现的呀?感谢感谢~~

**Comment by keyu-tian** [comment]
@YilanWang 可参考quant.py里面的相关插值代码

**Comment by YilanWang** [comment]
> @YilanWang 可参考quant.py里面的相关插值代码

感谢回复~,我就是发现quant有插值,在var里用了,但是vqvae.py好像没有使用欸,是直接拿1x32x16x16(suppose 输入是256)直接用的,包括我导出的onnx计算图也是这样 :) 

![Snipaste_2024-04-29_10-37-49](https://github.com/FoundationVision/VAR/assets/13708948/ead7cddd-ccec-487c-a563-b68ab8e836ee)

**Comment by keyu-tian** [comment]
> > @YilanWang 可参考quant.py里面的相关插值代码
> 
> 感谢回复~,我就是发现quant有插值,在var里用了,但是vqvae.py好像没有使用欸,是直接拿1x32x16x16(suppose 输入是256)直接用的,包括我导出的onnx计算图也是这样 :)
> 
> ![Snipaste_2024-04-29_10-37-49](https://private-user-images.githubusercontent.com/13708948/326318461-ead7cddd-ccec-487c-a563-b68ab8e836ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNzA0MDQsIm5iZiI6MTcxNDM3MDEwNCwicGF0aCI6Ii8xMzcwODk0OC8zMjYzMTg0NjEtZWFkN2NkZGQtY2NlYy00ODdjLWE1NjMtYjY4YWI4ZTgzNmVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDA1NTUwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3ZmJmZmRmNTQxNTdkMDdmYTEzZDFiYTY4MTgzZjkwODA5NmI0M2U0ZmExMmIwZjc0M2EzOTU3ZWM5ZmVhMDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XZYdaETQkGgsSCC_X1f5gD9x9lAHuNF6cRSdCjDW-mU)
> 
> @keyu-tian

嗯嗯，参考代码在什么地方调用了就ok
